### DISCUSSION TITLE
DataGrid: Set foreground color of specific cell in selected row

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#11345] [General]
How do I set the foreground (text) color of a specific cell in a selected row of a DataGrid?

I can set the color regardless selection with:

           
              <DataGridTextColumn Header="ThisOne" Binding="{Binding Value}" Foreground="Red"/>

But that sets the color for the entire column, both for selected and unselected rows.

I can also set the foreground of the entire selected row with:

            <DataGrid.Styles>
              <Style Selector="DataGridCell:selected">
                <Setter Property="Foreground" Value="Red"/>
              </Style>
            </DataGrid.Styles>

But that applies to all cells in the row, not just a specific one.

**Comment by maxkatz6** [comment]
See https://docs.avaloniaui.net/docs/styling/troubleshooting#selector-with-a-pseudoclass-doesnt-override-the-default

In your case, it's the easiest to look into default styles and copy+change
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L361-L363

**Comment by wgraaf** [comment]
Thanks for the reply.
Could you help me understand how to single out and set the foreground color for the cell in a particular column of a selected row? 
(say cell in column "ThisOne" of selected row?)

**Reply by timunie** [reply]
For example: 

```xml
<DataGrid ItemsSource="{Binding Countried}">
  <DataGrid.Styles>
    <Style Selector="DataGridRow:selected DataGridCell.Test">
      <Setter Property="Foreground" Value="Red" />
    </Style>
  </DataGrid.Styles>
  <DataGrid.Columns>
    <DataGridTextColumn Header="Country" 
                        CellStyleClasses="Test"
                        Binding="{Binding Name}" 
                        x:DataType="local:Country" />
...
```

**Reply by wgraaf** [reply]
Using a CellStyleClass was the missing ingredient.
Thanks!
